src/SFML/Graphics/Shape.cpp

```diff
@@ -31,16 +31,22 @@
 
 #include <algorithm>
 
+#include <cassert>
+#include <cmath>
+
 namespace
 {
-// Compute the normal of a segment
-sf::Vector2f computeNormal(const sf::Vector2f& p1, const sf::Vector2f& p2)
+// Compute the direction and normal unit vectors of a segment
+std::pair<sf::Vector2f, sf::Vector2f> computeDirectionAndNormal(const sf::Vector2f& p1, const sf::Vector2f& p2, bool flipped)
```

Comment:
I'm not particularly a fan of boolean parameters. I think they can exist if they have a major impact on the calculation and it would cost a lot maintenance-wise to duplicate the code, but here we're just flipping one of the returned results.

So the consumer of the function defines the output of the function, which it just as well could do on the consuming end of the function.

If anything, it should have at least a default value for `flipped`.

Reply:
I made code simple and removed the boolean parameter :+1: